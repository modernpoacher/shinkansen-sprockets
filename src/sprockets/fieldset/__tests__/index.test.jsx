import React from 'react'
import renderer from 'react-test-renderer'

import classnames from 'classnames'

import Super from '#sprockets/sprockets'
import Sprocket from '#sprockets/sprockets/fieldset'

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

jest.mock('#sprockets/sprockets/fieldset/title')
jest.mock('#sprockets/sprockets/fieldset/group')

describe('#sprockets/sprockets/fieldset', () => {
  describe('<Sprocket />', () => {
    describe('With required props', () => {
      const component = (
        <Sprocket />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(Sprocket.prototype.getClassName)
            .toBeDefined()
        })
      })

      describe('`renderTitle`', () => {
        it('is defined', () => {
          return expect(Sprocket.prototype.renderTitle)
            .toBeDefined()
        })
      })

      describe('`renderDescription`', () => {
        it('is defined', () => {
          return expect(Sprocket.prototype.renderDescription)
            .toBeDefined()
        })
      })

      describe('`renderErrorMessage`', () => {
        it('is defined', () => {
          return expect(Sprocket.prototype.renderErrorMessage)
            .toBeDefined()
        })
      })

      describe('`renderGroup`', () => {
        it('is defined', () => {
          return expect(Sprocket.prototype.renderGroup)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Sprocket
            title='MOCK TITLE'
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      let returnValue

      beforeEach(() => {
        jest.spyOn(Super.prototype, 'getClassName').mockReturnValue('MOCK GETCLASSNAME')

        const component = (
          <Sprocket />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        returnValue = instance.getClassName()
      })

      it('invokes `classnames`', () => {
        return expect(classnames)
          .toBeCalledWith('MOCK GETCLASSNAME', { error: false }, 'fieldset')
      })

      it('returns the classname', () => {
        return expect(returnValue)
          .toBe('MOCK CLASSNAME')
      })
    })
  })
})
