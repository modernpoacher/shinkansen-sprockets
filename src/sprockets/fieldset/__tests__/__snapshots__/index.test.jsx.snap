// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#sprockets/sprockets/fieldset <Sprocket /> With additional props renders 1`] = `
<div
  className="MOCK CLASSNAME"
>
  <fieldset
    className="MOCK CLASSNAME"
    onChange={[MockFunction]}
  >
    <legend
      className="MOCK CLASSNAME"
    >
      <span
        className="text-content"
      >
        MOCK TITLE
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`#sprockets/sprockets/fieldset <Sprocket /> With required props renders 1`] = `
<div
  className="MOCK CLASSNAME"
>
  <fieldset
    className="MOCK CLASSNAME"
    onChange={[Function]}
  />
</div>
`;
